File: src/main/java/com/pubnub/api/models/server/objects_api/EntityArrayEnvelope.java
package com.pubnub.api.models.server.objects_api;

import lombok.Getter;

import java.util.List;

@Getter
public class EntityArrayEnvelope<T> extends EntityEnvelope<List<T>> {

    protected Integer totalCount;
    protected String next;
    protected String prev;
}

File: src/main/kotlin/com/pubnub/api/models/server/objects_api/EntityArrayEnvelope.kt
package com.pubnub.api.models.server.objects_api

data class EntityArrayEnvelope<T>(
    val status: Int = 0,
    val data: Collection<T> = listOf(),
    val totalCount: Int? = null,
    val next: String? = null,
    val prev: String? = null
)
